app/app/Http/Controllers/VotingListsController.php

```diff
@@ -122,6 +122,40 @@ public function csv(VotingList $votingList)
         return response($csv, 200, ['Content-Type' => 'text/csv']);
     }
 
+    public function json(VotingList $votingList)
+    {
+        $members = $this->members($votingList)->map(function ($member) {
+            return [
+                'member_id' => $member->id,
+                'last_name' => $member->last_name,
+                'first_name' => $member->first_name,
+                'group' => [                
+                    'abbreviation' => $member->group->abbreviation,
+                    'name' => $member->group->name,
+                ],
+                'country' => [
+                    'code' => $member->country->value,
+                    'name' => $member->country->label,
+                ],
+                'position' => $member->pivot->position->label,
+                ];
+        })->toArray();
+
+        $voting= [
+            'title' => $votingList->display_title,
+            'date' => $votingList->date,
+            'result' => $votingList->result->label,
+            'members' => $members,
+        ];
+
+
+        ob_start();
+
+        $json = json_encode($voting);
+
+        return response($json, 200, ['Content-Type' => 'application/json']);
```

Comment:
This works 100% as expected. Laravel does however provide a few little convenience helpers to return JSON responses which are a little more idiomatic. The helper will handle setting the correct content type headers and status code for you.

Relevant docs: https://laravel.com/docs/8.x/responses#json-responses

```suggestion
        return response()->json($voting);
```

Reply:
And another small thing: The terminology is a little confusing and I’ll add a small glossary section to the docs:

* `Voting`: How a single member voted, e.g. "David Sasolli voted FOR in vote 123"
* `Vote`: A single vote, containing information like the date and result (e.g. `ADOPTED`/`REJECTED` for a single vote. Belongs to a single `VotingList`.
* `VotingList`: Has many `Votings` and belongs to a single `Vote`.
* `Vote`s and `VotingList`s basically represent the same "real-life object", but are stored in separate database tables, because we source them from different parliamentary records that become available at different times.
* `VoteCollection`: A collection of many related `Vote`s.

Reply:
So in this case the variable `$voting` actually is a `$vote` or `$votingList`.

Reply:
@tillprochaska those comments actually do exist, but we could debate migrating them into the readme or something like that. We explicitly decided against that earlier. @Alexander-Dubrawski you can find (hopefully helpful) descriptions for all models in the corresponding php files in the model folder. :)

Reply:
It is always helpful to have a description in the corresponding folders and files :). I also would find a diagram like for example:

![class-diagram](https://user-images.githubusercontent.com/32880898/140788103-bc42e5ce-6c6f-45a9-9325-af7f50a477da.png)

helpful. This does not need to be directly inside the readme, but maybe in a wiki? 🙃